Source/Core/Core/HW/WiimoteReal/IOWin.cpp

```diff
@@ -623,7 +654,18 @@ int IORead(HANDLE& dev_handle, OVERLAPPED& hid_overlap_read, u8* buf, int index)
     }
   }
 
-  return bytes + 1;
+  // ReadFile will always return 22 bytes read.
+  // So we need to calculate the acutal report size by its report id
+  DWORD report_size = static_cast<DWORD>(GetReportSize(buf[1]));
+  if (report_size == 0)
+  {
+    WARN_LOG(WIIMOTE, "Recieved unsupported report %d in Wii Remote %i", buf[1], index + 1);
+    return -1;
+  }
+
+  // 1 Byte for the Data Report Byte, another for the Report ID and the actual report size
+  bytes = std::min(bytes, report_size + 1) + 1;
```

Comment:
Good catch. Unless the OS/Driver messes up...but that is highly unlikely. And if so the memory is still valid, it just contains garbage/old data.